editor/src/clj/editor/gui.clj

```diff
@@ -1597,6 +1622,55 @@
 
+(g/defnode MaterialNode
+  (inherits outline/OutlineNode)
+  (property name g/Str
+            (dynamic error (g/fnk [_node-id name name-counts] (prop-unique-id-error _node-id :name name name-counts "Name")))
+            (set (partial update-gui-resource-references :material)))
+  (property child-index g/Int (dynamic visible (g/constantly false)) (default 0))
+
+  (property material resource/Resource
+            (value (gu/passthrough material-resource))
+            (set (fn [evaluation-context self old-value new-value]
+                   (project/resource-setter
+                     evaluation-context self old-value new-value
+                     [:resource :material-resource]
+                     [:shader :override-material-shader]
+                     [:build-targets :dep-build-targets])))
+            (dynamic edit-type (g/constantly
+                                 {:type resource/Resource
+                                  :ext ["material"]})))
+
+  (input name-counts NameCounts)
+
+  (input material-resource resource/Resource)
+
+  (input override-material-shader ShaderLifecycle :substitute (constantly nil))
+
+  (output node-id+child-index NodeIndex (g/fnk [_node-id child-index] [_node-id child-index]))
+  (output name+child-index NameIndex (g/fnk [name child-index] [name child-index]))
+  (output node-outline outline/OutlineData :cached (g/fnk [_node-id name child-index build-errors]
+                                                     {:node-id _node-id
+                                                      :node-outline-key name
+                                                      :label name
+                                                      :icon layer-icon
+                                                      :child-index child-index
+                                                      :outline-error? (g/error-fatal? build-errors)}))
+  (output resource-names GuiResourceNames (g/fnk [name] (sorted-set name)))
+  (input dep-build-targets g/Any)
+  (output dep-build-targets g/Any (gu/passthrough dep-build-targets))
+  (output pb-msg g/Any (g/fnk [name material-resource]
+                         {:name name
+                          :material (resource/resource->proj-path material-resource)}))
+  (output material-shaders GuiResourceShaders :cached (g/fnk [override-material-shader name]
+                                                    ;; If the referenced material-resource is missing, we don't return an entry.
+                                                    (when (some? override-material-shader)
+                                                      {name override-material-shader})))
+  (output material-names GuiResourceNames (g/fnk [name] (sorted-set name)))
+  (output build-errors g/Any (g/fnk [_node-id name name-counts]
+                               (g/package-errors _node-id
+                                                 (prop-unique-id-error _node-id :name name name-counts "Name")))))
```

Comment:
We should add a call to `(validate-material-resource _node-id material material-shader)` here like we have for the default material specified in the `GuiSceneNode` to make it a build error to use materials with array textures.

Reply:
Fixed